src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseTestCaseService.java

```diff
@@ -155,6 +155,9 @@ public boolean generateTestCasesFromFeedbacks(List<Feedback> feedbacks, Programm
         testCasesToSave.addAll(newTestCases);
         testCasesToSave.addAll(testCasesWithUpdatedActivation);
 
+        // Ensure no duplicate TestCase is present
+        testCasesToSave.removeIf(testCase -> testCasesToSave.stream().filter(t -> t.getTestName().equals(testCase.getTestName())).count() > 1);
```

Comment:
```suggestion
        testCasesToSave.removeIf(candidate -> testCasesToSave.stream().filter(testCase -> testCase.getTestName().equals(candidate.getTestName())).count() > 1);
```
Single character variable names should be avoided.